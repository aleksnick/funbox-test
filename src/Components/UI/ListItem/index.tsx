import React from "react";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import Paper from "@material-ui/core/Paper";
import ICanStyled from "../../../Models/ICanStyled";

export interface ListProps extends ICanStyled {
  extraContent?: React.ReactNode;
}

/**
 * Элемент списка
 *
 * @export
 * @class List
 * @extends {React.Component<ListProps>}
 */
export default class List extends React.Component<ListProps> {
  constructor(props: ListProps) {
    super(props);
  }

  render() {
    const { className, children, extraContent } = this.props;
    return (
      <Paper className={className}>
        <ListItem>
          <ListItemText primary={children} />
          <ListItemSecondaryAction>{extraContent}</ListItemSecondaryAction>
        </ListItem>
      </Paper>
    );
  }
}
