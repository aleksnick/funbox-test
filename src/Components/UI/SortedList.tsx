import React from "react";
import {
  DragDropContext,
  Droppable,
  Draggable,
  DropResult
} from "react-beautiful-dnd";

export interface DraggableItemProps {}

export interface SortedListProps {
  onSort?: (draggableId: number, destination: number) => void;
}

/**
 * Badge
 *
 * @export
 * @class SortedList
 * @extends {React.Component<SortedListProps>}
 */
export default class SortedList extends React.Component<SortedListProps> {
  constructor(props: SortedListProps) {
    super(props);
  }

  render() {
    const { children } = this.props;
    const childs = children as React.ReactElement<DraggableItemProps>[];

    return (
      <DragDropContext onDragEnd={this.onDragEnd}>
        <Droppable droppableId="droppable">
          {(provided, snapshot) => (
            <div ref={provided.innerRef}>
              {childs.map(
                (item: React.ReactElement<DraggableItemProps>, index) => {
                  const key = item.key || index;
                  return (
                    <Draggable
                      key={key}
                      draggableId={key.toString()}
                      index={index}
                    >
                      {(provided, snapshot) => (
                        <div
                          ref={provided.innerRef}
                          {...provided.draggableProps}
                          {...provided.dragHandleProps}
                        >
                          {item}
                        </div>
                      )}
                    </Draggable>
                  );
                }
              )}
              {provided.placeholder}
            </div>
          )}
        </Droppable>
      </DragDropContext>
    );
  }

  onDragEnd = (result: DropResult) => {
    if (!result || !result.destination) {
      return;
    }
    const onSort = this.props.onSort;
    if (onSort) {
      onSort(parseInt(result.draggableId), result.destination.index);
    }
  };
}
