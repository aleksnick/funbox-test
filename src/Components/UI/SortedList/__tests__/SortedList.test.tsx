import React from "react";
import SortedList from "../";
import ListItem from "../../ListItem";
import { render } from "enzyme";

describe("SortedList", () => {
  describe("Snapshots", () => {
    test("Simple SortedList snapshot", () => {
      const wrapper = render(
        <SortedList>
          <ListItem key={0}>One</ListItem>
          <ListItem key={1}>Two</ListItem>
          <ListItem key={2}>Three</ListItem>
        </SortedList>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
