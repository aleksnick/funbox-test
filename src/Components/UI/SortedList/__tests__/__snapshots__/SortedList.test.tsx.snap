// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortedList Snapshots Simple SortedList snapshot 1`] = `
<div>
  <div
    aria-roledescription="Draggable item. Press space bar to lift"
    data-react-beautiful-dnd-drag-handle="0"
    data-react-beautiful-dnd-draggable="0"
    draggable="false"
    tabindex="0"
  >
    <div
      class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
    >
      <li
        class="MuiListItem-container-29"
      >
        <div
          class="MuiListItem-root-28 MuiListItem-default-31 MuiListItem-gutters-36 MuiListItem-secondaryAction-38"
        >
          <div
            class="MuiListItemText-root-40"
          >
            <span
              class="MuiTypography-root-46 MuiTypography-subheading-53 MuiListItemText-primary-43"
            >
              One
            </span>
          </div>
        </div>
        <div
          class="MuiListItemSecondaryAction-root-82"
        />
      </li>
    </div>
  </div>
  <div
    aria-roledescription="Draggable item. Press space bar to lift"
    data-react-beautiful-dnd-drag-handle="0"
    data-react-beautiful-dnd-draggable="0"
    draggable="false"
    tabindex="0"
  >
    <div
      class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
    >
      <li
        class="MuiListItem-container-29"
      >
        <div
          class="MuiListItem-root-28 MuiListItem-default-31 MuiListItem-gutters-36 MuiListItem-secondaryAction-38"
        >
          <div
            class="MuiListItemText-root-40"
          >
            <span
              class="MuiTypography-root-46 MuiTypography-subheading-53 MuiListItemText-primary-43"
            >
              Two
            </span>
          </div>
        </div>
        <div
          class="MuiListItemSecondaryAction-root-82"
        />
      </li>
    </div>
  </div>
  <div
    aria-roledescription="Draggable item. Press space bar to lift"
    data-react-beautiful-dnd-drag-handle="0"
    data-react-beautiful-dnd-draggable="0"
    draggable="false"
    tabindex="0"
  >
    <div
      class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
    >
      <li
        class="MuiListItem-container-29"
      >
        <div
          class="MuiListItem-root-28 MuiListItem-default-31 MuiListItem-gutters-36 MuiListItem-secondaryAction-38"
        >
          <div
            class="MuiListItemText-root-40"
          >
            <span
              class="MuiTypography-root-46 MuiTypography-subheading-53 MuiListItemText-primary-43"
            >
              Three
            </span>
          </div>
        </div>
        <div
          class="MuiListItemSecondaryAction-root-82"
        />
      </li>
    </div>
  </div>
</div>
`;
